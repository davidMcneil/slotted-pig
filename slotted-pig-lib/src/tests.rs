use anyhow::Result;
use insta::assert_json_snapshot;
use test_case::test_case;

use crate::{categorizer::Categorizer, transaction::Transaction};

#[test_case("tests/categorizer_empty.json", "tests/transactions_empty.csv", "empty"; "empty")]
#[test_case("tests/categorizer_simple.json", "tests/transactions_simple.csv", "simple"; "simple")]
fn test_categorizer(categorizer: &str, transactions: &str, name: &str) -> Result<()> {
    let categorizer = Categorizer::from_json_file(categorizer)?;
    let transactions = Transaction::from_csv_file(transactions)?;
    assert_json_snapshot!(name, categorizer.categorize(&transactions));
    Ok(())
}
