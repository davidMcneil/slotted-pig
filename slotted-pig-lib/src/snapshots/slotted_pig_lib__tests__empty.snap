---
source: slotted-pig-lib/src/tests.rs
expression: categorizer.categorize(&transactions)
---
{
  "categorized": []
}
